datafusion/optimizer/src/filter_push_down.rs

```diff
@@ -89,10 +90,11 @@ fn get_predicates<'a>(
         .filters
         .iter()
         .filter(|(_, columns)| {
-            !columns
-                .intersection(used_columns)
-                .collect::<HashSet<_>>()
-                .is_empty()
+            columns.is_empty()
+                | !columns
```

Comment:
Somewhat more optimal to use `||` here to avoid computing  the right arm.

Reply:
Nice catch!